mesonbuild/interpreter/type_checking.py

```diff
@@ -853,3 +856,59 @@ def _pkgconfig_define_convertor(x: T.List[str]) -> PkgConfigDefineType:
     default=[],
     convertor=_pkgconfig_define_convertor,
 )
+
+def _dep_fallback_feat_validator(val: T.List[str]) -> T.Iterable[FeatureCheckBase]:
+    if len(val) == 1:
+        yield FeatureNew('feature as a string or array of one element', '0.54.0', 'use the two element form')
+
+_USE_CONFIG_TOOL = "Use the 'config-tool' method instead"
+
+
+DEPENDENCY_KWS: T.List[KwargInfo] = [
+    KwargInfo('allow_fallback', (bool, NoneType), since='0.56.0'),
+    KwargInfo('cmake_args', ContainerTypeInfo(list, str), default=[], listify=True, since='0.50.0'),
+    KwargInfo('cmake_module_path', ContainerTypeInfo(list, str), default=[], listify=True, since='0.50.0'),
+    KwargInfo('cmake_package_version', str, default='', since='0.57.0'),
+    KwargInfo('components', ContainerTypeInfo(list, str), default=[], listify=True, since='0.54.0'),
+    DEFAULT_OPTIONS.evolve(since='0.38.0'),
+    DISABLER_KW.evolve(since='0.49.0'),
+    KwargInfo(
+        'fallback',
+        (ContainerTypeInfo(list, str), NoneType),
+        listify=True,
+        validator=lambda x: 'Must be a string, or an array of 0, 1, or 2 elements' if len(x) > 2 else None,
+        feature_validator=_dep_fallback_feat_validator,
+    ),
+    KwargInfo(
+        'include_type',
+        str,
+        default='preserve',
+        validator=in_set_validator({'preserve', 'system', 'non-system'}),
```

Comment:
This list is repeated 3 times: here, in `as_system_method`, and in the `IncludeTypes` litteral. I wonder if you could define it only once?

Reply:
The Literal unfortunately I don't think there is a way to help :/ in the other case if I ever get around to adding something like KwargInfo for typed_pos_args we could probably get rid of